src/components/HomePage/Slideshow/Slideshow.js

```diff
@@ -0,0 +1,241 @@
+    const { index } = this.state;
+
+    this.setState({ isClicked: true });
+    i != index && this.updateIndex('', i);
+  };
+
+  deleteAutoPlay = () => {
+    this.setState({
+      autoPlay: false,
+    });
+
+    clearInterval(this.sliderInterval);
+  };
+
+  startSlideShow = () => {
+    this.sliderInterval = setInterval(this.nextSlide, 4000);
+  };
+
+  onMouseLeftHandler = () => {
+    const { isClicked } = this.state;
+
+    !isClicked &&
+      this.setState({
+        autoPlay: true,
+      });
+  };
+
+  transitionEnd = () => {
+    const { index } = this.state;
+    const { slideData } = this.props;
+    const slideDataLength = slideData.length;
+    const indexOfLastCloneSlide = -1;
+    const indexOfFirstCloneSlide = slideDataLength;
+    const indexOfLastSlide = slideDataLength - 1;
+
+    index === indexOfFirstCloneSlide &&
+      this.setState({
+        index: 0,
+        shouldTransition: false,
+      });
+
+    index === indexOfLastCloneSlide &&
+      this.setState({
+        index: indexOfLastSlide,
+        shouldTransition: false,
+      });
+  };
+
+  removeTransform = () => {
+    this.setState({
+      isTransform: false,
+    });
+  };
+
+  componentDidMount() {
+    this.startSlideShow();
+  }
+
+  componentDidUpdate(prevProps, { autoPlay: prevStateAutoPlay }) {
+    const { autoPlay } = this.state;
+
+    if (autoPlay && autoPlay != prevStateAutoPlay) {
+      this.startSlideShow();
+    }
+  }
+
+  render() {
+    const { index, prevIndex, isTransform, shouldTransition } = this.state;
+    const { slideData, animation } = this.props;
+    const getPaginationClassNames = (slideIndex) =>
+      cx(`${CN}-pagination__item`, {
+        [`${CN}-pagination__item--active`]: index === slideIndex,
+      });
+    const pagination = slideData.map((el, i) => (
+      <div
+        key={i}
+        className={getPaginationClassNames(i)}
+        onClick={(e) => this.changeSlideWithPagination(e, i)}
+      ></div>
+    ));
+
+    const styleSlide = {
+      transform: `translateX(${100 *
+        (index - prevIndex)}%)`,
+    };
+
+    const slides = slideData.map((el, i) => (
+      <Slide
+        key={el.id}
+        bgImage={el.img}
+        style={
+          prevIndex === i && isTransform ? styleSlide : {}
+        }
+        animation={cx({
+          [animation]:
+            this.isAnimationZoom() && i === prevIndex,
+          [FADE]: animation === FADE && i === index,
+        })}
+        onAnimationEnd={this.removeTransform}
+      >
+        {' '}
+        {el.component}
+      </Slide>
+    ));
+
+    const {
+      0: firstCloneData,
+      length,
+      [length - 1]: lastCloneData,
+    } = slideData;
+
+    const firstSlideClone = (
+      <Slide key="firstClone" bgImage={firstCloneData.img}>
+        {firstCloneData.component}
+      </Slide>
+    );
+
+    const lastSlideClone = (
+      <Slide key="lastClone" bgImage={lastCloneData.img}>
+        {lastCloneData.component}
+      </Slide>
+    );
```

Comment:
Is there any reason use const for these markups ?

Reply:
No reason